### PR TITLE
Update and implementation of UMP SDK to 2.2.0/2.4.0

### PR DESCRIPTION
**Pull request by LTPhantom**
## Description

Updated the UMP SDK version for Android and iOS and surfaced the missing methods on the dart layer. (GMA SDK for iOS already comes with the latest UMP SDK)

## Related Issues

* https://github.com/googleads/googleads-mobile-flutter/issues/944

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by malandr2** [comment]
ACK: https://github.com/googleads/googleads-mobile-flutter/issues/1024

**Comment by Reprevise** [comment]
I definitely do appreciate a more Dart-like approach with these new APIs. Could there also be a `loadAndShowPrivacyOptionsFormIfRequired`? (maybe too long of a name)

File: packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m [hunk 1]
```diff
@@ -57,6 +57,25 @@ - (void)handleMethodCall:(FlutterMethodCall *_Nonnull)call
     UMPConsentStatus status =
         UMPConsentInformation.sharedInstance.consentStatus;
     result([[NSNumber alloc] initWithInteger:status]);
+  } else if ([call.method
+                 isEqualToString:@"ConsentInformation#canRequestAds"]) {
+    result(@([UMPConsentInformation.sharedInstance canRequestAds]));
+  } else if ([call.method
+                 isEqualToString:@"ConsentInformation#"
+                                 @"getPrivacyOptionsRequirementStatus"]) {
```
Comment (jjliu15) [inline packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m:65]: Should this be one line?

File: packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m [hunk 2]
```diff
@@ -73,6 +92,24 @@ - (void)handleMethodCall:(FlutterMethodCall *_Nonnull)call
                                            details:error.domain]);
                                }
                              }];
+  } else if ([call.method
+                 isEqualToString:@"UserMessagingPlatform#"
```
Comment (jjliu15) [inline packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m:96]: Same here

**Review by jjliu15 — COMMENTED** [review]
Can you add a note in the PR description that the iOS GMA version already has the required UMP version?

File: packages/google_mobile_ads/lib/src/ump/consent_information.dart [hunk 3]
```diff
@@ -61,6 +61,24 @@ abstract class ConsentInformation {
   /// the platform API has been called.
   Future<void> reset();
 
+  /// Indicates whether the app has completed the necessary steps for gathering updated user consent.
+  Future<bool> canRequestAds();
```
Comment (malandr2) [inline packages/google_mobile_ads/lib/src/ump/consent_information.dart:65]: Can this just be a `bool`? Or does it have to be `Future<bool>`?

File: packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m (hunk 1 above)
Comment (LTPhantom) [reply packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m:65]: Format tools changed it to two lines.

File: packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m (hunk 2 above)
Comment (LTPhantom) [reply packages/google_mobile_ads/ios/Classes/UserMessagingPlatform/FLTUserMessagingPlatformManager.m:96]: Same as above

File: packages/google_mobile_ads/lib/src/ump/consent_information.dart (hunk 3 above)
Comment (LTPhantom) [reply packages/google_mobile_ads/lib/src/ump/consent_information.dart:65]: It does have to be a `Future<bool>`. The returned values from platform calls are async.

File: packages/google_mobile_ads/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+## Next Version
+* Adds missing methods from the latest UMP SDK version.
```
[the remark this answers is not in the document]
Comment (LTPhantom) [reply packages/google_mobile_ads/CHANGELOG.md]: Done

**Comment by malandr2** [comment]
> I definitely do appreciate a more Dart-like approach with these new APIs. Could there also be a `loadAndShowPrivacyOptionsFormIfRequired`? (maybe too long of a name)

There is a `showPrivacyOptionsForm()` API that can be triggered from a button in your app settings, for example. You can query `getPrivacyOptionsRequirementStatus()` to determine if you need to show this button or not.

**Comment by malandr2** [comment]
@LTPhantom iOS recently updated their UMP SDK version to `2.4.0` last week

**Comment by LTPhantom** [comment]
> @LTPhantom iOS recently updated their UMP SDK version to `2.4.0` last week

Acknowledged. Current code in this PR should be compatible with it.

**Review by jjliu15 — APPROVED** [review]

File: packages/google_mobile_ads/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## Next Version
+* Adds support for methods from the [Android](https://developers.google.com/admob/android/privacy/release-notes) UMP SDK version 2.2.0.
+* Adds support for methods from the [iOS](https://developers.google.com/admob/ios/privacy/download#release_notes) UMP SDK version 2.3.0.
```
Comment (malandr2) [inline packages/google_mobile_ads/CHANGELOG.md:3]: 2.4.0

File: packages/google_mobile_ads/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
+## Next Version
+* Adds support for methods from the [Android](https://developers.google.com/admob/android/privacy/release-notes) UMP SDK version 2.2.0.
```
Comment (malandr2) [inline packages/google_mobile_ads/CHANGELOG.md:2]: Nit: `APIs` instead of `methods`

**Review by malandr2 — APPROVED** [review]

**Pull request merged**